### DISCUSSION TITLE
How do you stop the Web Worker in the Editor?

### DISCUSSION THREAD

**Discussion by WiFi77** [pyscript/pyscript#2215] [Q&A] [answered]
I can’t find a way to terminate the Web Worker for an editor instance. I have a React component that creates an editor, and when this component unmounts, I can remove the script, but the Web Worker remains in the browser’s memory. Over time, they accumulate. This isn’t an issue with the terminal, but it is with the editor.

**Comment by WebReflection** [comment]
@WiFi77 would it make sense to expose an `xworker` able to **terminate()** the worker, same way you would terminate a terminal? Please note that won't be necessarily created out of the box (it's lazy loaded on first run) so you'd need to `script.xworker?.terminate()` eventually, and that means the editor won't be able to run anymore. This is a feature asking for troubles though but if you *really* know what you are doing, I think we should provide a way to terminate the worker or **maybe** we could provide a `kill()` method to that script that will automatically destroy the editor too, but that might be a bit too obtrusive if we remove nodes out of the blue.

Thoughts? /cc @ntoll

**Reply by WebReflection** [reply, marked as answer]
OK ... I went ahead and provided this MR https://github.com/pyscript/pyscript/pull/2216 ... if giving you a way to grab the `xworker` (which is just a `Worker` instance) and explicitly `.terminate()` it works for you, we should be good next release. I might publish that in *npm* first so you could play around ... would that be OK?

**Reply by WebReflection** [reply]
It's up and running as `https://cdn.jsdelivr.net/npm/@pyscript/core@0.6.4/dist/core.js`

**Reply by WiFi77** [reply]
Thank you, that’s exactly what I needed! I have several editor instances on multiple subpages of my React application, so being able to terminate the Web Worker like this is really helpful.
